app/commands/thredded/autofollow_mentioned_users.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+module Thredded
+  class AutofollowMentionedUsers
+    def initialize(post)
+      @post = post
+    end
+
+    def run
+      members = autofollows
+      return unless members.present?
```

Comment:
I'm unsure whether this return needs to exist here because the work has already been done.

If I'm reading this right, we'd want the guard clause at the beginning to ensure that we're not doing any subsequent work *if* there are no members present - is that right? If so, what about:

```ruby
def run
  return if members.empty?
  autofollow_members
end

private

def members
  user_names = Thredded::AtNotificationExtractor.new(post.content).run
  members = post.readers_from_user_names(user_names).to_a
  members.delete(post.user)
  members = exclude_those_opting_out_of_at_notifications(members)
  members
end

def autofollow_members
  Thredded::UserTopicFollow.create_unique(
    user.id,
    post.postable_id,
    Thredded::UserTopicFollow::REASON_MENTIONED
  )
end